paf-mvp-operator-express/src/operator-node.ts

```diff
@@ -240,11 +241,11 @@ export class OperatorNode extends Node {
       this.beginHandling,
       timeout(this.redirectResponseTimeoutInMs),
       this.checkQueryString,
-      this.checkReturnUrl,
       this.checkReadPermission,
       this.checkReadIdsAndPreferencesSignature,
+      this.checkReturnUrl,
```

Comment:
This check is now done after permission and signature verification, so that we can save the return URL when we know we can trust it.